module2/intermission_work/sql.md

```diff
@@ -0,0 +1,71 @@
+---
+title: SQL Intermission Work
+layout: page
+---
+
+SQL is a programming language used to interact with a database. Learning to interact with databases is crucial to your work as a backend developer. While we eventually want to use libraries that will write SQL for us, it is important to be familiar with how SQL works under the hood so that you understand what your code is doing at the database level.
+
+For your SQl intermission work, you will complete an introductory tutorial, and then will complete some exercises to help you practice the most important SQL concepts. This assignment is meant as an introduction to sql, so don't be alarmed if you don't feel like you have mastered these concepts. Use the accompanying Checks for Understanding to assess yourself.
+
+## Jumpstart Lab Tutorial
+
+Work through the [Jumpstart Lab SQL Tutorial](http://tutorials.jumpstartlab.com/topics/sql/fundamental_sql.html), **but do not complete the "Sequel" tutorial linked at the bottom of the Jumpstart Lab page.** Then complete the Checks for Understanding.
+
+### Checks for Understanding
+
+1. What is a database?
+1. What is SQL?
+1. What is SQLite3?
+1. What is a Table?
+1. What is a primary key?
+1. What is a foreign key?
+1. Explain what each of the following SQL commands do:
+  * insert
+  * select
+  * where
+  * order by
+  * inner join
+
+## SQL Exercises
+
+Now you'll work through some of the exercises at [pgexercises.com](https://pgexercises.com/). In the Jumpstart Lab tutorial, you used SQLite3. In these exercises, you will use Postgresql. It's largely the same, but make sure that when you are looking at documentation you are looking at the [Postgresql Documentation](https://www.postgresql.org/docs/9.3/index.html).
+
+Work through the three following sections (Basic, Joins, Aggregates) of exercises on the site. Make sure you are also reading the Answers and Discussion section at the bottom, even if you get the answer right. There are some helpful explanations and alternative solutions.
+
+To reiterate from before, you do not need to be a master of SQL coming out these exercises, so you should **timebox these exercises**. Timebox each of the three sections to **no more than 2 hours** for a maximum of 6 hours spent on these exercises. There will likely be exercises that you have difficulty completing, and that's okay! Use the provided hints, give it your best effort, and skip to the answer if you can't get it. Just make sure you read and understand the answer before moving on.
```

Comment:
I'm not totally sure if this is the chopped version or if there used to be even more, but my preference is to chop even more. We have so many different sections in this intermission work, I think this SQL section should really be the fundamental fundamentals. 

I would stick with what's needed for these learning goals:


1. What is a database?
1. What is SQL?
1. What is SQLite3?
1. What is a Table?
1. What is a primary key?
1. What is a foreign key?
1. Explain what each of the following SQL commands do:
  * insert
  * select
  * where
  * join
  
  So No Multiple Joins and no Aggregates at all and maybe not even unique and order.

Reply:
Actually let's also keep order_by and distinct as learning goals because they are needed for the active record lesson.

Reply:
Agreed we should chop the multiple joins, aggregates, grouping etc. I also think we could drop the idea of joining and foreign keys at all cause it won't be introduced until week 4/5. KEep it simple. They can just think about one table.